week2/homework/exercises/1-blog-API/server.js

```diff
@@ -1,10 +1,87 @@
+'use strict'
 const express = require('express')
 const app = express();
- 
+app.use(express.json());
 
+const fs = require("fs");
+const { title } = require('process');
+ 
 // YOUR CODE GOES IN HERE
-app.get('/', function (req, res) {
-  res.send('Hello World')
-})
+
+app.get('/blogs/:title', (req, res)  => {readBlogs(req, res);})
+
+app.post('/blogs' , (req, res) => {createBlogs(req, res);})
+
+app.put('/posts/:title' , (req, res) => {updateBlogs(req, res);})
+
+app.delete('/blogs/:title' , (req, res) => {deleteBlogs(req, res);})
+
+//delete 
+function deleteBlogs(req, res){
+  const title = req.params.title;
+  if (fs.existsSync(title)){
+    fs.unlinkSync(title);
+    res.end('ok')
+  } else {
+    res.send('This post does not exist!')
+  }
+}
+
+//update
+function updateBlogs(req, res){
+  if (isInvalid(req)) {
+    res.status(400);
+    res.send('invalid request');
+    return;
+  }
+  let updateBlogs = {
+    title : req.body.title,
+    content : req.body.content
+  }
+  const title = req.params.title;
+   if (fs.existsSync(title)){
+   fs.writeFileSync(title, JSON.stringify(updateBlogs));
+   res.send('ok')
+   } else {
+     res.send('This post does not exist!')
+   }
+}
+    
+//create
+function createBlogs(req, res){
+  if (isInvalid(req)){
+    res.status(400);
+    res.send('Invalid request');
+    return;
+  }
+  let newBlog = {
+    title: req.body.title,
+    content : req.body.content,
+  }
+   fs.writeFileSync('blogs.json', JSON.stringify(newBlog));
```

Comment:
The idea is that each blog post goes to a separate file. This is not the best design, but then again data is usually stored in a database which you will learn about in the next course.
```suggestion
   fs.writeFileSync(newBlog,title, JSON.stringify(newBlog));
```